infrastructure-playbooks/purge-cluster.yml

```diff
@@ -419,51 +340,61 @@
-    when: ceph_block_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph block"
     failed_when: false
     register: ceph_block_partition_to_erase_path
 
   - name: get ceph journal partitions
     shell: |
-      blkid | awk '/ceph journal/ { sub (":", "", $1); print $1 }'
-    when: ceph_journal_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph journal"
     failed_when: false
     register: ceph_journal_partition_to_erase_path
 
   - name: get ceph db partitions
     shell: |
-      blkid | awk '/ceph block.db/ { sub (":", "", $1); print $1 }'
-    when: ceph_db_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph block.db"
     failed_when: false
     register: ceph_db_partition_to_erase_path
 
   - name: get ceph wal partitions
     shell: |
-      blkid | awk '/ceph block.wal/ { sub (":", "", $1); print $1 }'
-    when: ceph_wal_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph block.wal"
     failed_when: false
     register: ceph_wal_partition_to_erase_path
 
+  - name: set_fact combined_devices_list
+    set_fact:
+      combined_devices_list: "{{ ceph_data_partition_to_erase_path.stdout_lines +
+                                 ceph_lockbox_partition_to_erase_path.stdout_lines +
+                                 ceph_block_partition_to_erase_path.stdout_lines +
+                                 ceph_journal_partition_to_erase_path.stdout_lines +
+                                 ceph_db_partition_to_erase_path.stdout_lines +
+                                 ceph_wal_partition_to_erase_path.stdout_lines }}"
+
+  - name: resolve parent device
+    shell: echo /dev/$(lsblk -no pkname "{{ item }}")
+    register: tmp_resolved_parent_device
+    with_items:
+      - "{{ combined_devices_list }}"
+
+  - name: set_fact resolved_parent_device
+    set_fact:
+      resolved_parent_device: "{{ tmp_resolved_parent_device.results | map(attribute='stdout') | list | unique }}"
+
   - name: zap ceph journal/block db/block wal partitions
     shell: |
       # if the disk passed is a raw device AND the boot system disk
-      if echo "{{ item }}" | egrep -sq '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p){1,2}$' && parted -s $(echo "{{ item }}" | egrep -o '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p){1,2}') print | grep -sq boot; then
+      if parted -s "{{ item }}" print | grep -sq boot; then
         echo "Looks like {{ item }} has a boot partition,"
         echo "if you want to delete specific partitions point to the partition instead of the raw device"
         echo "Do not use your system disk!"
         exit 1
       fi
-      raw_device=$(echo "{{ item }}" | egrep -o '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]|nvme[0-9]n[0-9]){1,2}')
-      partition_nb=$(echo "{{ item }}" | egrep -o '[0-9]{1,2}$')
-      sgdisk --delete $partition_nb $raw_device
+      sgdisk -Z "{{ item }}"
+      dd if=/dev/zero of="{{ item }}" bs=1M count=200
```

Comment:
200MB is huge. 10MB should be more than enough I guess.

Reply:
if it's only for time-consuming concern, it doesn't really matter since it's pretty insignificant :
```209715200 bytes (210 MB) copied, 0.187801 s, 1.1 GB/s```
 but we can stick to 10M if you think it's better anyway